### ISSUE TITLE
No Reservations Found: Shutting down Resy Booking Bot

### ISSUE THREAD

**Issue by somewhereinnyc** [Alkaar/resy-booking-bot#82]
Hi there, the restaurant I am trying to get reservations for randomly opens reservations from 10 AM to 10:30 AM every morning and I set the bot up to snipe at 10 AM. However, they have not opened reservations yet and the bot thinks it missed. Is there a way to keep the bot continually going and waiting for the reservations to open?

Thank you, sending you a coffee later today for making this awesome program!

<img width="785" alt="CleanShot 2023-02-07 at 10 39 34@2x" src="https://user-images.githubusercontent.com/124703673/217291636-a1439c3a-4680-4658-9a55-dabf4042481a.png">

**Comment by Corgasaurus** [comment]
So if it is the restaurant that starts with a C, I just went and changed it minute by minute until it triggered. The few times I tried it, it was 10:15.

**Comment by somewhereinnyc** [comment]
That is the restaurant and yeah it seems like 10:15 is the time but missed the shot again.

**Comment by Corgasaurus** [comment]
They're hip to what is going for sure. I just stepped it up minute by minute until it triggered. I noticed this last night, and I don't think it was there when I actually triggered a reservation. I'm doing one for a friend and while we both have cards on I file, idk if that is their bot deterrent.  ![image](https://user-images.githubusercontent.com/111210497/217899144-d1d46ce9-73a7-4d25-ab7e-9d99c62de38e.jpeg)

**Comment by somewhereinnyc** [comment]
Not sure if bot deterrent but I am missing every day for the times and seating I want. Curious if the bot can start doing time ranges like 10 AM to 10:15 AM etc

**Comment by Alkaar** [comment]
Just to clarify the error message in the [first screenshot](https://user-images.githubusercontent.com/124703673/217291636-a1439c3a-4680-4658-9a55-dabf4042481a.png), `Could not find any available reservations` means there was no reservation found that overlaps with the reservations in your requested list. It does NOT mean there are no reservations at all, just none that match your search criteria.

> So if it is the restaurant that starts with a C, I just went and changed it minute by minute until it triggered. The few times I tried it, it was 10:15.

I would be curious what restaurant this is. I tried looking at restaurants of a similar name that I'm familiar with but it seems like this is not publicized on the site. I have mentioned in the past why didn't want to implement this 👉 https://github.com/Alkaar/resy-booking-bot/issues/52#issuecomment-1272680346. However, if it becomes more common, then I may have no choice. It's not hard to do, it was more of a matter of principle.

-----

Regarding the [second screenshot](https://user-images.githubusercontent.com/111210497/217899144-d1d46ce9-73a7-4d25-ab7e-9d99c62de38e.jpeg), I think it's addressing a few things. It's probably to reduce cancellations but also the reselling of reservations which some sites offer. It's impossible for a normal person to get a reservation which is why I wrote this bot but I didn't think it was fair to "profit" off it. While bots themselves are a bit of a gray area, I thought the "fairest" way to go about it is to offer it to free to others which is why this is public.

I think reservations becoming available at different times is more to combat bots. As I said above, if it becomes more commonplace, I may consider updating the bot for this use case.

**Comment by somewhereinnyc** [comment]
Hi @Alkaar this is what happened to me today reserving this restaurant.

It tried booking twice, went through payment token and missed twice. So it basically got snipped before the payment token could go through. I wish I had the screenshot but it was two long token payment ID's it showed and missed each time.

**Comment by Alkaar** [comment]
For restaurants that I've tried this does happen from time to time. Not much can be done here if someone else snipes the same spot from under you. There are also cases where it tells you the booking failed but actually, it succeeded. I can't fix this because a bad response is a bad response. I don't really want to second-guess the return response as that's a bit awkward.

**Comment by somewhereinnyc** [comment]
Yeah this just might be the hardest restaurant to get into, 4 fails in a row from quicker people. Honestly just trying to get a spot to take my girlfriend for our anniversary. Will keep trying! Thank you again for the help!
